tensorly/tenalg/einsum_tenalg/caching.py

```diff
@@ -0,0 +1,20 @@
+EINSUM_PATH_CACHE = dict()
+
+
+def einsum_path_cached(fun):
+    def wrapped(key, *args, **kwargs):
+        name = fun.__name__
```

Comment:
Should we use

```suggestion
name = f"{fun.__module__}.{fun.__name__}"
```

instead to make sure we don't introduce any bugs if we have two functions with the same name?

Reply:
That's a good idea.